### PR TITLE
Unrevert Receipt Center and patch popover bugs

### PR DESCRIPTION
**Pull request by YodaLightsabr**
Changes since the initial merge of #3953:
https://github.com/hackclub/bank/compare/3935.receipt-pairing-ui...ian/receipt-center-patch

This PR fixes:
- [Renaming a transaction inside of a modal](https://hackclub.slack.com/archives/C047Y01MHJQ/p1686688754110489?thread_ts=1686683829.606249&cid=C047Y01MHJQ)
- [A bug where `event` is `nil`](https://heroku-app97991095.airbrake.io/projects/288439/groups/3559654844659365182)
  - These were caused by the popover code expecting an event on the HcbCode that isn’t there, and since we’re not showing transaction details for mock data anyways, it’s safe to just disable the popovers on playground events.
  - code
- [A bug where the `text` was not properly passed into `modal_header`](https://heroku-app97991095.airbrake.io/projects/288439/groups/3559607160640598478?tab=overview)

### CODE REVIEW

**Comment by malted** [comment]
I think my playground mode PR holds the record for most bugs. King of problematic pulls right here.

File: app/models/hcb_code.rb [hunk 1]
```diff
@@ -152,20 +169,20 @@ def events
...
+      partner_donation: ["Partner donation", " to "],
+      ach: ["ACH", " in "],
+      check: ["Check", " in "],
+      disbursement: ["Disbursement", " to "],
+      card_charge: ["Card charge", " in "],
+      bank_fee: ["Bank Fee", " in "]
     }[type || :unknown]
 
-    event ? "#{pretty_type} #{event.name}" : pretty_type[0..-4]
+    event&.name && show_event_name ? pretty_type.join("") + event.name : pretty_type.first
```
Comment (garyhtou) [inline app/models/hcb_code.rb]:
```suggestion
  def pretty_title(show_event_name: true)
    pretty_type = {
      unknown: ["Transaction", "in"],
      invoice: ["Invoice", "in"],
      donation: ["Donation", "to"],
      partner_donation: ["Partner donation", "to"],
      ach: ["ACH", "in"],
      check: ["Check", "in"],
      disbursement: ["Disbursement", "to"],
      card_charge: ["Card charge", "in"],
      bank_fee: ["Bank Fee", "in"]
    }[type || :unknown]
    
    return pretty_type.first unless event&.name && show_event_name

    (pretty_type + [event.name]).join(" ")
```

please test this first!

File: db/schema.rb
Comment (garyhtou) [inline db/schema.rb:1]: Please remove the unrelated changes (`payment_for` and `exports` table)

File: app/views/canonical_pending_transactions/_canonical_pending_transaction.html.erb [hunk 2]
```diff
@@ -194,3 +195,80 @@
     <% end %>
   <% end %>
 </tr>
+
+<% if pt.local_hcb_code.event.present? && pt.local_hcb_code.pretty_title.present? %>
```
Comment (garyhtou) [inline app/views/canonical_pending_transactions/_canonical_pending_transaction.html.erb]: Just glancing at this, I'm not sure if the `pt.local_hcb_code.pretty_title.present?` is necessary.

If it's not needed, remove it from `ct` too

File: app/views/canonical_pending_transactions/_canonical_pending_transaction.html.erb [hunk 3]
```diff
@@ -194,3 +195,80 @@
...
+                        unknown: "Transaction",
+                        invoice: "Invoice",
+                        donation: "Donation",
+                        partner_donation: "Partner donation",
+                        ach: "ACH",
+                        check: "Check",
+                        disbursement: "Disbursememt",
+                        card_charge: "Card charge",
+                        bank_fee: "Bank Fee"
+                      }[pt.local_hcb_code.type] %>
```
Comment (garyhtou) [inline app/views/canonical_pending_transactions/_canonical_pending_transaction.html.erb]: Can we do `pretty_title(show_event_name: false)` here?

File: app/views/canonical_transactions/_canonical_transaction.html.erb [hunk 4]
```diff
@@ -212,3 +213,80 @@
...
+                        unknown: "Transaction",
+                        invoice: "Invoice",
+                        donation: "Donation",
+                        partner_donation: "Partner donation",
+                        ach: "ACH",
+                        check: "Check",
+                        disbursement: "Disbursememt",
+                        card_charge: "Card charge",
+                        bank_fee: "Bank Fee"
+                      }[ct.local_hcb_code.type] %>
```
Comment (garyhtou) [inline app/views/canonical_transactions/_canonical_transaction.html.erb]: Can we do `pretty_title(show_event_name: false)` here?

**Review by garyhtou — APPROVED** [review]
@YodaLightsabr and I heavily tested this in staging. There are a couple of UI quirks. But all the functionality is there, and it doesn't cause any 500 errors!
